browser/extensions/api/brave_rewards_api.cc

```diff
@@ -605,7 +605,9 @@ ExtensionFunction::ResponseAction BraveRewardsSaveAdsSettingFunction::Run() {
   AdsService* ads_service_ = AdsServiceFactory::GetForProfile(profile);
   if (ads_service_) {
     if (params->key == "adsEnabled") {
-      ads_service_->SetEnabled(params->value == "true");
+      const auto is_enabled =
+          params->value == "true" && ads_service_->IsSupportedLocale();
```

Comment:
The bug was caused by this line, due to a legacy issue where the ads flag was enabled even if the users locale was unsupported